health_check/contrib/beat_health_check/backends.py

```diff
@@ -0,0 +1,100 @@
+from datetime import timedelta
+from typing import Any, Dict, List, Union
+
+from celery.beat import Service
+from celery.schedules import crontab, solar, schedule
+from django.conf import settings
+from django.utils.module_loading import import_string
+from django_celery_beat.models import CrontabSchedule, IntervalSchedule, SolarSchedule
+from django_celery_beat.schedulers import ModelEntry
+from health_check.backends import BaseHealthCheckBackend
+from health_check.exceptions import ServiceReturnedUnexpectedResult, ServiceUnavailable
+
+
+class CeleryBeatHealthCheck(BaseHealthCheckBackend):
+    def check_status(self):
+        """
+        Checks for overdue tasks in a celery beat scheduler.
+        Uses the scheduler module dotted path that is specified in settings.py with `CELERY_BEAT_SCHEDULER`.
+        If not specified, defaults to `django_celery_beat`'s `django_celery_beat.schedulers.DatabaseScheduler`.
+        Allows a custom buffer to be set using `BEAT_HEALTH_CHECK_BUFFER_SECONDS` in settings.py. The buffer
+        defaults to 30 seconds if not defined. The buffer will offset the scheduler interval for when due
+        tasks are processed. Using a buffer avoids false positives, such as the case where a task is
+        technically due according to the scheduler, but that's only because the scheduler has not hit its
+        interval to check and process due tasks.
+        """
+        from celery.app import default_app as celery_app
```

Comment:
Let's use the `current_app` proxy to get the currently used app, not the default one (fallback one)
Also, I think (but haven't tested it) that this import can be moved to the root module imports section because `current_app` is just a proxy.